test/functional/test_runner.py

```diff
@@ -27,6 +27,15 @@
 import logging
 import unittest
 
+# Should fail testing and immediately stop if free space falls below.
+# This would be checked before running any and every test.
+# TODO consider increasing to 16 * 1024 * 1024 (https://github.com/bitcoin/bitcoin/pull/23100#discussion_r716825510)
```

Comment:
why add a todo?

Reply:
because you wrote you think a test requires 16M

a todo is the best I can add here about that

Reply:
I think it would be better to change the number instead of adding a TODO to change the number

Reply:
I think so as well, but I can't personally commit it

Reply:
why?

Reply:
because I don't trust you, because I don't even know you,
because one should only commit code one can take ownership of,
because I can't take ownership of an assertion you seem not
to be even sure of,
because I would lie by committing it!

Reply:
You don't have to trust me. The value is in the source code `BLOCKFILE_CHUNK_SIZE`, or you can test it yourself: https://github.com/bitcoin/bitcoin/pull/23100#discussion_r716825510

Even if wasn't, what is the worst thing that could happen by increasing the number?

Reply:
> The value is in the source code `BLOCKFILE_CHUNK_SIZE`, or you can test it yourself

I would have a look, thanks.

I was hoping that many of the >200 functional tests would not strictly require that a large space reserve. 🙄

> Even if wasn't, what is the worst thing that could happen by increasing the number?

In case the constant didn't apply to all tests, it would result on unnecessary test halting for maybe successfully passable tests, and even maybe aborting a test run able to end successfully.

Reply:
I have had a casual look,
I'd say there are many functional tests ending and being removed
upon success way before reaching 16MiB,
I'm feeling like removing MIN_FREE_SPACE_BEFORE_TEST. 🤔

Reply:
There's even some functional tests occupying only KiBs and in a blink of an eye it's successful and gone.

In a few days I'd be removing MIN_FREE_SPACE_BEFORE_TEST.